liboptv/tests/check_segmentation.c

```diff
@@ -6,62 +6,73 @@
 
 
 START_TEST(test_peak_fit)
-{
-    int ntargets; 
-    unsigned char img[5][5] = {
+    {
+    int i,j, ntargets;
+    unsigned char tmpimg[5][5] = {
         { 0,   0,   0,   0, 0},
         { 0, 255, 255, 255, 0},
         { 0, 255, 255, 255, 0},
         { 0, 255, 255, 255, 0},
         { 0,   0,   0,   0, 0}
     };
-    
+
+
     target pix[1024];
-    
+
     control_par cpar = {
         .imx = 5,
         .imy = 5,
-    }; 
+    };
 
+    unsigned char* img;
+    img = calloc(cpar.imx*cpar.imy, sizeof(unsigned char));
+    for (i=0;i<cpar.imy;i++) for(j=0;j<cpar.imx;j++) {img[i*cpar.imx+j]=tmpimg[i][j];}
```

Comment:
And if you really want a copy, then ``memcpy()``. Or at least, space out the writing.